### ISSUE TITLE
Feature request: support case-insensitive directory match

### ISSUE THREAD

**Issue by kergoth** [Tarrasch/zsh-bd#12]
This is supported by https://github.com/vigneshwaranr/bd with an argument in its alias: "To enable case-insensitive directory name matching, use -si instead of -s in the alias.". Of course, one could enable case-insensitive tab-completion in general and use the zsh-bd tab-completion for this, but one might not want it enabled everywhere.

**Comment by Tarrasch** [comment]
> but one might not want it enabled everywhere.

Is this actually the case for you? Why would you not want this in general? Look how useful it is. :)

![selection_031](https://cloud.githubusercontent.com/assets/294349/6149412/f9101acc-b208-11e4-8896-32a009a35237.png)

**Comment by Tarrasch** [comment]
Of course, if you want to implement it and add tests for this and make it not the default behavior. I'll happily merge it. Same goes for #11. :)

**Comment by kergoth** [comment]
Fair enough, thanks for the quick responses.

**Comment by Tarrasch** [comment]
Thanks for opening these issues. I'm happy you care for the best of the project. :)

**Comment by tom-auger** [comment]
I spent some time working out how to enable case-insensitive matching just for bd, so posting in case it's useful for others.
To enable case-insensitive matches change the following line in `bd.zsh`:

```
compctl -V directories -K _bd bd
```
to
```
compctl -M 'm:{a-zA-Z}={A-Za-z} r:|=* l:|=* r:|=*' -V directories -K _bd bd
```
The -M parameter lets you pass a custom match specification for auto completion. The match spec here is the same one used to enable global case-insensitive matching in oh-my-zsh.
